roles/kubespray-defaults/defaults/main.yaml

```diff
@@ -477,7 +477,11 @@ kube_apiserver_endpoint: |-
   {% if loadbalancer_apiserver is defined -%}
       https://{{ apiserver_loadbalancer_domain_name }}:{{ loadbalancer_apiserver.port|default(kube_apiserver_port) }}
   {%- elif not is_kube_master and loadbalancer_apiserver_localhost -%}
+    {%- if loadbalancer_apiserver_type == "dns-hosts" and loadbalancer_apiserver_localhost -%}
+      kubernetes.default.svc.{{ dns_domain }}:{{ kube_apiserver_port }}
```

Comment:
kubernetes.default - this is predefined service name.
This service created by api-server automticaly

Don't use it elsewhere.
Better to use another unique name
For example use `kubernetes-api.dns-round-robin-balancer-via-hosts-file.local`